Vegetarian-Food-Finder/Controller/SearchViewController.swift

```diff
@@ -54,19 +53,21 @@ final class SearchViewController: UIViewController, UITableViewDataSource, UITab
     }
 
     func updateSearchResults(for searchController: UISearchController) {
-        let searchBar = searchController.searchBar
-        let searchText = searchBar.text!
-        filterForTextSearch(searchText: searchText)
+        tableView.reloadData()
     }
 
-    func filterForTextSearch(searchText: String) {
-        filteredRestaurantsList = restaurantList.filter({ Business in
-            if searchController.searchBar.text != "" {
-                let searchTextMatch = Business.name.lowercased().contains(searchText.lowercased())
-                return searchTextMatch
-            }
-            return true
-        })
-        tableView.reloadData()
+    func filterForTextSearch(searchText: String) -> [Business] {
+        if searchText == "" {
+            return restaurantList
+        } else {
+            let filteredRestaurantsList = restaurantList.filter({ Business in
```

Comment:
You don't need to store the filtered RestaurantsList. You can directly return the filtered list from the filter function